content/docs/engine-dev/architecture/prefabs/_index.md

```diff
@@ -0,0 +1,645 @@
+
+
+### Prefab instance
+<!-- <a name="prefab-instance"> -->
+
+A *prefab instance* defines a prefab object that represents the DOM stored in the prefab templates. Similar to the template DOM, the prefab instance has member variables for the container entity, entities and nested instances. There is a one-to-many relationship between a prefab template and prefab instances.
+
+For example, if there is a Car prefab on file, there will be one Car template defined. Many instances can be generated using that single car template. Prefab instances can be populated by deserializing DOM into instance objects using `InstanceSerializer::Load()` method. Prefab instances can also be built from the ground up by adding entities to them directly. Similar to templates, prefab instances can be converted into DOM format by using `InstanceSerializer::Store()` method. Prefab serialization and deserialization will be discussed in detail in a following section.
+
+![Prefab-Instance Relationship](./images/PrefabInstance.png)
+
+| Code | Description |
+| :-   | :-          |
+| [Instance](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.h) | Class that represents the instance objects instantiated from prefab templates. |
+| [InstanceSerializer](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceSerializer.cpp) | Class that loads DOMs into prefab instances and stores prefab instances into DOMs. |
+
+
+### Prefab link
+<!-- <a name="prefab-link"> -->
+
+A *prefab link* defines how two templates are connected. A link exists between two templates, identified by a `LinkId` (a unique integer like `TemplateId`) and defined by a source template id and a target template id. The target template is the one which has a nested instance in it that points to a source template.
+
+```
+Bike       (Bike.prefab)
+| Wheel    (Wheel.prefab)
+```
+
+In the above example of link between Bike and Wheel templates, the target template would be the one corresponding to `Bike.prefab` file and the source template would be the one corresponding to `Wheel.prefab` file. When the target template is loaded, the prefab system component creates a link between the two relevant templates and maintains a mapping of link ids to link objects.
+
+One of the key members of a link object is the link DOM, which contains a file path to the source template and a list of patches to be applied on the source template. A link DOM may or may not contain patches in it. In the case where there are no patches, the target template would simply fetch the template DOM of the source template.
+
+> **Note**
+> In the latest development branch, a link maintains a `PrefabOverridePrefixTree` rather than a link DOM. The link DOM will be generated from the tree on the fly. This supports the new **Prefab Override** feature.
```

Comment:
```suggestion
{{ <note> }}
In the latest development branch, a link maintains a `PrefabOverridePrefixTree` rather than a link DOM. The link DOM will be generated from the tree on the fly. This supports the new **Prefab Override** feature.
{{ </note> }}
```